R/checks.R

```diff
@@ -96,8 +96,9 @@ check_stan_delay <- function(dist) {
       "and infinite maximum."
     )
   }
+  assert_numeric(attr(dist, "tolerance"), lower = 0, upper = 1)
   # Check that `dist` has a finite maximum
-  if (any(is.infinite(max(dist)))) {
+  if (any(is.infinite(max(dist))) && !(attr(dist, "tolerance") > 0)) {
     stop("All distribution passed to the model need to have a finite maximum")
```

Comment:
I think this error message can be improved. For example, "If tolerance is set to 0, all distributions must have a finite `max`."

```r
> delay_opts(LogNormal(mean = 4, sd = 1), tolerance = 0)
Error in check_stan_delay(dist) : 
  All distribution passed to the model need to have a finite maximum
```

Reply:
Have reworded to "All distribution passed to the model need to have a finite maximum, which can be achieved either by setting `max` or non-zero `tolerance`."